core/src/main/java/org/elasticsearch/gateway/PrimaryShardAllocator.java

```diff
@@ -48,15 +50,30 @@
  */
 public abstract class PrimaryShardAllocator extends AbstractComponent {
 
-    @Deprecated
-    public static final String INDEX_RECOVERY_INITIAL_SHARDS = "index.recovery.initial_shards";
+    private static final Function<String, String> INITIAL_SHARDS_PARSER = (value) -> {
```

Comment:
It'd be nice if this got a class the same way AutoExpandReplicas did. Not required at all, but pleasant.